packages/contracts/src/contracts/current/protocol/Exchange/MixinSignatureValidator.sol

```diff
@@ -131,20 +147,55 @@ contract MixinSignatureValidator is
             isValid = signer == recovered;
             return isValid;
 
-        // Signature using EIP712
-        } else if (signatureType == SignatureType.EIP712) {
+        // Implicitly signed by caller.
+        // The signer has initiated the call. In the case of non-contract
+        // accounts it means the transaction itself was signed.
+        // Example: let's say for a particular operation three signatures
+        // A, B and C are required. To submit the transaction, A and B can
+        // give a signature to C, who can then submit the transaction using
+        // `Caller` for his own signature. Or A and C can sign and B can
+        // submit using `Caller`. Having `Caller` allows this flexibility.
+        } else if (signatureType == SignatureType.Caller) {
             require(
-                signature.length == 66,
+                signature.length == 0,
                 INVALID_SIGNATURE_LENGTH
             );
-            v = uint8(signature[1]);
-            r = readBytes32(signature, 2);
-            s = readBytes32(signature, 34);
-            recovered = ecrecover(hash, v, r, s);
-            isValid = signer == recovered;
+            isValid = signer == msg.sender;
             return isValid;
 
-        // Signature from Trezor hardware wallet
+        // Signature verified by signer contract.
+        // If used with an order, the maker of the order is the signer contract.
+        } else if (signatureType == SignatureType.Signer) {
+            isValid = ISigner(signer).isValidSignature(hash, signature);
+            return isValid;
```

Comment:
Pretty much any of our signature types could be implemented in a validator, but I think it's still worth keeping the most used ones as defaults for the sake of efficiency. 

One other question I'm thinking about is how to prevent people from removing their validator's approval as a griefing mechanism. For this reason, I think it's still preferable that we keep as many defaults as possible.